operator/src/main/java/org/keycloak/operator/v2alpha1/KeycloakDeployment.java

```diff
@@ -16,82 +16,162 @@
+            distConfig.putAll(keycloakCR.getSpec().getDistConfig());
+        }
+
+        container.setEnv(distConfig.entrySet().stream()
+                .map(e -> new EnvVarBuilder().withName(e.getKey()).withValue(e.getValue()).build())
+                .collect(Collectors.toList()));
+
+//        Set<String> configSecretsNames = new HashSet<>();
+//        List<EnvVar> configEnvVars = distConfig.entrySet().stream()
+//                .map(e -> {
+//                    EnvVarBuilder builder = new EnvVarBuilder().withName(e.getKey());
+//                    Matcher matcher = CONFIG_SECRET_PATTERN.matcher(e.getValue());
+//                    // check if given config var is actually a secret reference
+//                    if (matcher.matches()) {
+//                        builder.withValueFrom(
+//                                new EnvVarSourceBuilder()
+//                                        .withNewSecretKeyRef(matcher.group(2), matcher.group(1), false)
+//                                        .build());
+//                        configSecretsNames.add(matcher.group(1)); // for watching it later
+//                    } else {
+//                        builder.withValue(e.getValue());
+//                    }
+//                    builder.withValue(e.getValue());
+//                    return builder.build();
+//                })
+//                .collect(Collectors.toList());
+//        container.setEnv(configEnvVars);
+//        this.configSecretsNames = Collections.unmodifiableSet(configSecretsNames);
+//        Log.infof("Found config secrets names: %s", configSecretsNames);
+
+        return baseDeployment;
     }
 
-    public Deployment newKeycloakDeployment(Keycloak keycloak) {
-        if (baseDeployment == null) {
-            URL url = this.getClass().getResource("/base-deployment.yaml");
-            baseDeployment = client.apps().deployments().load(url).get();
+    public void updateStatus(KeycloakStatusBuilder status) {
+        if (existingDeployment == null) {
+            status.addNotReadyMessage("No existing Deployment found, waiting for creating a new one");
+            return;
         }
 
-        var deployment = baseDeployment;
+        var replicaFailure = existingDeployment.getStatus().getConditions().stream()
+                .filter(d -> d.getType().equals("ReplicaFailure")).findFirst();
+        if (replicaFailure.isPresent()) {
+            status.addNotReadyMessage("Deployment failures");
+            status.addErrorMessage("Deployment failure: " + replicaFailure.get());
+            return;
+        }
 
-        deployment
-                .getSpec()
-                .setReplicas(keycloak.getSpec().getInstances());
+        if (existingDeployment.getStatus() == null
+                || existingDeployment.getStatus().getReadyReplicas() == null
+                || existingDeployment.getStatus().getReadyReplicas() < keycloakCR.getSpec().getInstances()) {
```

Comment:
We might want to improve a little on this in the future with something on those lines:

```java
existingDeployment.getStatus().getReadyReplicas() == existingDeployment.getSpec().getReplicas() &&
existingDeployment.getStatus().getReadyReplicas() == keycloakCR.getSpec().getInstances() &&
existingDeployment.getStatus().getUnavailableReplicas() == 0
```
but is good enough for now.

Reply:
Do we care about unavailable replicas and all replicas? I think the only thing that concerns us is the number of ready replicas.

Reply:
not fussy on this, totally good enough especially for the baseline 👍